GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Client/TestRailClientWrapper.cs

```diff
@@ -17,33 +21,58 @@ public class TestRailClientWrapper
         private static readonly Logger Log = LogManager.GetLogger(MethodBase.GetCurrentMethod()?.DeclaringType?.Name);
         private readonly TestRailClient _testRailClient;
 
+        private int? _requestsCount;
+        private readonly Config _config;
+        private readonly int _attemptsCount;
+        private readonly int _sleepDuration;
+
+        public int? RequestsCount
+        {
+            get => _requestsCount;
+            set
+            {
+                if (value < 0) 
+                    throw new ArgumentException("Number of requests per minute must be positive");
+                _requestsCount = value;
+            }
+        }
+
         public TestRailClientWrapper(TestRailClient testRailClient)
         {
             _testRailClient = testRailClient;
+            _config = ConfigurationManager.GetConfiguration();
+            RequestsCount ??= 0;
+            _attemptsCount = _config.TestRailRetriesCount ?? 3;
```

Comment:
It is like this or null checking, because TestRailClientWrapper does not know anything about checks that we make in the config - and the type of this variable is `nullable int`

Reply:
Let's think about how to keep only config and default values. Because in this case we have default in a two places.